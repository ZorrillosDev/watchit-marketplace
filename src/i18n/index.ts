import i18n from "i18next";
import {initReactI18next} from "react-i18next";
import LanguageDetector from "i18next-browser-languagedetector";

// Languages
import en from './langs/en.json'
import es from './langs/es.json'

// default lang
export const defaultLang = 'en';

// Translation
const resources = {
    en: {translation: en},
    es: {translation: es}
}


i18n
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
        resources,
        fallbackLng: defaultLang,
        interpolation: {
            escapeValue: false
        }
    })

// Humanize displaying
export const lang = [
    {value: "en", name: "English"},
    {value: "es", name: "Spanish"}

]

export default i18n;
