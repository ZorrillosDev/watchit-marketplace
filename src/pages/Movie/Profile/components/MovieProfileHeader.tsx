// REACT IMPORTS
import React, { FC } from 'react'

// MUI IMPORTS
import { Avatar, Button, ButtonProps, Grid, styled, Theme, Typography, Zoom } from '@mui/material'

// THIRD PARTY
import { IconMovie, IconHeart, IconShare } from '@tabler/icons'

// PROJECT IMPORTS
import { Fingerprint, MoreHoriz } from '@components/Icons'
import { LightTooltip } from '@components/Tooltip'
import { Translation } from '@src/i18n'
import { SxProps } from '@mui/system'

// ===========================|| MOVIE - PROFILE - HEADER ||=========================== //

const MovieProfileHeader: FC<{sx?: SxProps<Theme>}> = ({ sx }): JSX.Element => {
  return (
    <Grid item xs={12} display='flex' alignItems='center' justifyContent='space-between' sx={sx}>
      <LightTooltip TransitionComponent={Zoom} title={<Translation target='MOVIE_PROFILE_HEADER_FINGERPRINT' />}>
        <Avatar>
          <Fingerprint />
        </Avatar>
      </LightTooltip>
      <Grid container spacing={2} width='auto'>
        <Grid item>
          <MovieProfileButton
            size='small' sx={{ px: 2, py: '0.72rem', svg: { width: '1rem', height: '1rem' } }}
          >
            <IconHeart stroke={1.5} />
            <Typography sx={{ pl: 1 }}>3200</Typography>
          </MovieProfileButton>
        </Grid>
        <Grid item>
          <MovieProfileButton
            size='small'
            sx={{ px: 2, py: '0.72rem', svg: { width: '1rem', height: '1rem' } }}
          >
            <IconShare stroke={1.5} />
            <Typography sx={{ pl: 1 }}>
              <Translation target='MOVIE_PROFILE_HEADER_SHARE' />
            </Typography>
          </MovieProfileButton>
        </Grid>
        <Grid item>
          <MovieProfileButton size='small'>
            <IconMovie stroke={1} />
          </MovieProfileButton>
        </Grid>
        <Grid item>
          <MovieProfileButton size='small'>
            <MoreHoriz />
          </MovieProfileButton>
        </Grid>
      </Grid>
    </Grid>
  )
}

export default MovieProfileHeader

const MovieProfileButton = styled(Button)<ButtonProps>(({ theme }) => ({
  padding: '0.5rem',
  minWidth: 'auto',
  border: '1px solid',
  borderColor: theme.palette.divider,
  alignItems: 'center',
  justifyContent: 'center',
  [theme.breakpoints.down('md')]: {
    padding: '0.5rem !important',
    p: {
      display: 'none'
    },
    svg: {
      width: '24px !important',
      height: '24px !important'
    }
  }
}))
