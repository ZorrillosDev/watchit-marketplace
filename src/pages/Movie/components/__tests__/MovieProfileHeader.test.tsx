import React from 'react'
import { mount, shallow } from 'enzyme'
import MovieProfileHeader from '@pages/Movie/components/MovieProfileHeader'
import i18n from '@src/i18n'

/* eslint-disable  @typescript-eslint/consistent-type-assertions */

describe('<MovieProfileHeader />', () => {
  it('should render', () => {
    const component = shallow(<MovieProfileHeader />)

    expect(component).toMatchSnapshot()
  })

  it('should have movie profile header valid share text', () => {
    const translate = i18n.t('MOVIE_PROFILE_HEADER_SHARE')
    const component = mount(<MovieProfileHeader />)
    const componentText = component.find('button').at(1)

    expect(componentText.exists()).toBeTruthy()
    expect(componentText.text()).toContain(translate)
  })
})
