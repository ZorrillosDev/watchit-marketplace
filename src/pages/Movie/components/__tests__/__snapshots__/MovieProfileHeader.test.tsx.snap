// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieProfileHeader /> should render 1`] = `ShallowWrapper {}`;
